Reformatting, new test

chris-smith opened the pull request:
-Basic tests for message generation
-Discovered some very strange formatting courtesy of changing Atom settings partway through (unfortunately that is most of this change)

IanTheEngineer commented on example.js:
```diff
@@ -1,10 +1,11 @@
 
 let rosjs = require('./index.js');
+const std_msgs = rosjs.requires('std_msgs').msg;
+const ListCameras = rosjs.requires('baxter_core_msgs').srv.ListCameras;
```
Do you really want your test to depend on `baxter_core_msgs`? Perhaps something similar to the `add_two_ints` service example?

chris-smith replied:
switched to std_srvs/SetBool

IanTheEngineer commented on utils/message_utils.js:
```diff
@@ -26,57 +26,75 @@ let cmakePaths = cmakePath.split(':');
 // message serialization/deserialization
 //-----------------------------------------------------------------------
 
 let MessageUtils = {
```
Can you give a brief explanation (in comments) of how the message finding / caching / loading works for rosjs?

IanTheEngineer commented on utils/network_utils.js:
```diff
@@ -25,104 +25,104 @@ let MIN_PORT = 49152;
+  },
+
+  /**
    * FIXME: should this just return ROS_HOSTNAME
```
Should this FIXME for ROS_HOSTNAME be an issue?

chris-smith replied:
I don't think so currently - whether to use this or not is an option you can set when you initialize the node.

IanTheEngineer commented on utils/network_utils.js:
```diff
@@ -25,104 +25,104 @@ let MIN_PORT = 49152;
+          return;
+        }
+        //else
+        _freePortCheck(minPort, maxPort, callback);
```
Is there any timeout for this `freePortCheck()` or is it infinitely recursive?

chris-smith replied:
Not currently - it's assuming there's enough space that it will find one 'reasonably' quickly. That's also partially why it chooses a random port in the range instead of successively incrementing where it's more likely two things could discover 'hey this port is free' and then both try to use it. We do cover the case of 'this port was free but now its not' but clashes would occur more frequently.

IanTheEngineer commented on utils/serialization_utils.js:
```diff
@@ -30,22 +30,22 @@ const Transform = require('stream').Transform;
+  // buffer until we have the whole thing
+  this._messageBuffer = [];
 
   // FIXME: These are specific to parsing a service response...
```
Similarly, this should be an issue.

chris-smith replied:
FIXME here is maybe misleading... its more like a TODO. DeserializeStream is being used for any TCPROS deserialization, but the scheme is slightly different for services than for everything else so this is mostly me saying "Maybe don't conflate these things"

IanTheEngineer commented:
Just minor comments above. Aside from those, this looks great! :shipit:

IanTheEngineer commented on index.js:
```diff
@@ -111,20 +113,29 @@ let Rosjs = {
     });
   },
 
+  requires(msgPackage) {
```
This should be `require()` rather than `requires()`

chris-smith replied:
Thank you for including this comment for posterity

chris-smith committed:
Reformatting, tests, bugs

-Basic genjs tests
-Update msg include method
-Previously, rosjs would load all message
files it could find at init. Instead,
it will now find all the paths it expects to
include from and cache them. Message packages are
required through rosjs instead of through the
default 'requires' method according to these paths.
-Fixes npm tests for new require method
-Fixes bug on publisher disconnect
-Moves test.js to better-named example.js
